src/ngraph/ngraph_nnvm_ops.cc

```diff
@@ -77,6 +78,16 @@ void update_aux_vals(const std::shared_ptr<Graph> &graph,
   }
 }
 
+void compile_if_needed(std::shared_ptr<Graph> graph, int mode) {
+  if (mode == static_cast<int>(GraphExeMode::kTrain)) {
```

Comment:
Any reason to not use the `int(GraphExeMode::kTrain)`?  It strikes me as more succinct and a tad easier to read.

Reply:
Do you mean (int)(GraphExeMode::kTrain)? If so, it's less safe, in general, we shouldn't be using C-style casting.

Reply:
Nope, I really did mean `int(GraphExeMode::kTrain)`.

I'm not aware of any safety concerns when the operand is a enum-class literal.  But I wouldn't recommend holding up the PR just because of this either.

Reply:
...I'm not sure I've ever seen int as a function in C++.